compiler/rustc_infer/src/infer/fudge.rs

```diff
@@ -198,48 +198,43 @@ impl<'a, 'tcx> TypeFolder<'tcx> for InferenceFudger<'a, 'tcx> {
                     debug_assert!(
                         self.infcx.inner.borrow_mut().type_variables().probe(vid).is_unknown()
                     );
-                    ty
+                    Ok(ty)
                 }
             }
             ty::Infer(ty::InferTy::IntVar(vid)) => {
-                if self.int_vars.contains(&vid) {
-                    self.infcx.next_int_var()
-                } else {
-                    ty
-                }
+                Ok(if self.int_vars.contains(&vid) { self.infcx.next_int_var() } else { ty })
```

Comment:
Nit/idea: I'm wondering if this would look better as `self.int_vars.contains(&vid).then(|| self.infcx.next_int_var()).ok_or(ty)`

And same elsewhere with this pattern

Reply:
That's not the same... it should be `Ok` in both cases, whereas the suggested change would result in `Err` if `vid` is not in `int_vars`.

Reply:
Ah, right.